nb/projects/belt_sprockets/belt_sprockets.nb

```diff
@@ -0,0 +1,741 @@
+//Ported to JSxCAD from https://www.thingiverse.com/thing:16627/files by @BarbourSmith
+
+// Parametric Pulley with multiple belt profiles
+// by droftarts January 2012
+
+// Based on pulleys by:
+// http://www.thingiverse.com/thing:11256 by me!
+// https://github.com/prusajr/PrusaMendel by Josef Prusa
+// http://www.thingiverse.com/thing:3104 by GilesBathgate
+// http://www.thingiverse.com/thing:2079 by nophead
+
+// dxf tooth data from http://oem.cadregister.com/asp/PPOW_Entry.asp?company=915217&elementID=07807803/METRIC/URETH/WV0025/F
+// pulley diameter checked and modeled from data at http://www.sdp-si.com/D265/HTML/D265T016.html
+
+// tune-able constants
+
+//  ********************************
+//  ** Scaling tooth for good fit **
+//  ********************************
+/*  To improve fit of belt to pulley, set the following constant. Decrease or increase by 0.1mm at a time. We are modelling the *BELT* tooth here, not the tooth on the pulley. Increasing the number will *decrease* the pulley tooth size. Increasing the tooth width will also scale proportionately the tooth depth, to maintain the shape of the tooth, and increase how far into the pulley the tooth is indented. Can be negative */
+
+const additionalToothWidth = 0.2; //mm
+
+//  If you need more tooth depth than this provides, adjust the following constant. However, this will cause the shape of the tooth to change.
+
+const additionalToothDepth = 0; //mm
+
+// Functions
+
+const ToothSpaceingCurvefit = (teeth, b, c, d) => {
+  return ((c * Math.pow(teeth, d)) / (b + Math.pow(teeth, d))) * teeth;
+};
+
+const ToothSpacing = (teeth, tooth_pitch, pitch_line_offset) =>
+  2 * ((teeth * tooth_pitch) / (3.14159265 * 2) - pitch_line_offset);
+
+const ToothProfileToFace = (...points) =>
+  Shape.fromOpenPath(points).close().fill();
+
+const MXL = {
+  profile: ToothProfileToFace(
+    ...[
+      [-0.660421, -0.5, 0],
+      [-0.660421, 0, 0],
+      [-0.621898, 0.006033, 0],
+      [-0.587714, 0.023037, 0],
+      [-0.560056, 0.049424, 0],
+      [-0.541182, 0.083609, 0],
+      [-0.417357, 0.424392, 0],
+      [-0.398413, 0.458752, 0],
+      [-0.370649, 0.48514, 0],
+      [-0.336324, 0.502074, 0],
+      [-0.297744, 0.508035, 0],
+      [0.297744, 0.508035, 0],
+      [0.336268, 0.502074, 0],
+      [0.370452, 0.48514, 0],
+      [0.39811, 0.458752, 0],
+      [0.416983, 0.424392, 0],
+      [0.540808, 0.083609, 0],
+      [0.559752, 0.049424, 0],
+      [0.587516, 0.023037, 0],
+      [0.621841, 0.006033, 0],
+      [0.660421, 0, 0],
+      [0.660421, -0.5, 0],
+    ]
+  ),
```

Comment:
The only think I'd suggest is that it would be nice to be able to see what the profiles look like, so I'd use Polygon(...).gridView().md('MXL profile') instead of Polygon(...)

Reply:
👍 👍